presto-main/src/main/java/com/facebook/presto/operator/TableFinishOperator.java

```diff
@@ -156,12 +243,45 @@ public Page getOutput()
         return page.build();
     }
 
+    private List<ComputedStatistics> getComputedStatistics()
+    {
+        ImmutableList.Builder<ComputedStatistics> statistics = ImmutableList.builder();
+        while (!statisticsAggregation.isFinished()) {
+            Page page = statisticsAggregation.getOutput();
```

Comment:
`verify(statisticsAggregation.isBlocked().isDone());` to fail fast rather than voidly looping in case something goes wrong